test/jdk/java/net/SocketPermission/SocketPermissionIm.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * @test
+ * @bug 8243376
+ * @summary SocketPermission implies(Permission p) spec allows, If the object was initialized with a single IP address and one of p's IP addresses is equal to this object's IP addr
+ * @run java -Dsun.net.inetaddr.ttl=0 SocketPermissionIm
+ */
+
+import java.net.SocketPermission;
+import java.net.InetAddress;
+import java.io.*;
+import java.io.File;
+
+public class SocketPermissionIm {
+    public static void main(String[] args) throws Exception {
+        String hostname = "www.exmp.com";
+        String hostsFileName = System.getProperty("test.src", ".") + File.separator + "Host.txt";
+        System.setProperty("jdk.net.hosts.file", hostsFileName);
+
```

Comment:
It's better to specify this system property on the command line because (we discovered recently) there are some environments where the name service gets initialized before main() is called in which case setting the property here is too late. So, you need to add a -Djdk.net.hosts.file=${test.src}....

Reply:
Removed the system property and added this as a command-line option.